### ISSUE TITLE
protocol messages

### ISSUE THREAD

**Issue by stef** [cfrg/draft-irtf-cfrg-opaque#85]
in the specification there is a part which defines `{#protocol-messages}`:
```
enum {
    registration_request(1),
    registration_response(2),
    registration_upload(3),
    credential_request(4),
    credential_response(5),
    (255)
} ProtocolMessageType;

struct {
    ProtocolMessageType msg_type;    /* protocol message type */
    uint24 length;                   /* remaining bytes in message */
    select (ProtocolMessage.msg_type) {
        case registration_request: RegistrationRequest;
        case registration_response: RegistrationResponse;
        case registration_upload: RegistrationUpload;
        case credential_request: CredentialRequest;
        case credential_response: CredentialResponse;
    };
} ProtocolMessage;
```
in our implementation at https://github.com/stef/libopaque we omitted implementing this, for the following reasons:
  * this is depending on how an application is using this protocol,  it might not be necessary to wrap protocol messages in such frames, for example because the app already has it's own framing, or different ways to communicate this.
  * parsing any kind of input from possible adversaries should be done higher up the stack, where possibilities are richer to do so in memory safe ways or by implementing language security mechanisms to do so.

libopaque does however implement the credential and envelope structures.

**Comment by chris-wood** [comment]
Yeah, as you point out, this isn't strictly necessary to implement OPAQUE, so we ought to remove it from the spec. @klewi, objections?

**Comment by kevinlewi** [comment]
I am all for removing this!

**Comment by chris-wood** [comment]
Great! See #86 for the change.
